File: Programmers/week2/138476/jjunii/138476.py
def solution(k, tangerine):
    cnt_list = [0] * (max(tangerine) + 1)
    for size in tangerine:
        cnt_list[size] += 1
        
    cnt_list.sort(reverse = True)
    
    answer = 0
    for cnt in cnt_list:
        answer += 1
        k -= cnt
        if k <= 0:
            return answer

File: Programmers/week2/138476/kwonjin2/138476.js
function solution(k, tangerine) {
  const count = {};

  for (let i = 0; i < tangerine.length; i++) {
    count[tangerine[i]] = (count[tangerine[i]] || 0) + 1;
  }

  const counts = Object.values(count).sort((a, b) => b - a);

  let sum = 0;
  let kinds = 0;

  for (let c of counts) {
    sum += c;
    kinds++;
    if (sum >= k) break;
  }

  return kinds;
}
